---
title: "Pathogen Diagnostics using Nanopore MinIon"
subtitle: "Oxford Nanopore sequencing technology as a rapid, portable and accurate tool for fungal pathogen diagnostics in chickpea"
author: "Ido Bar & Oak Hatzimanolis"
date: "17 May 2019"
always_allow_html: yes
output: 
    # md_document:
#      css: "style/style.css"
      # toc: true
      # toc_depth: 3
#      highlight: pygments
#      number_sections: false
    html_document:
      css: "style/style.css"
      toc: true
      toc_float: true
      toc_depth: 3
      highlight: pygments
      number_sections: false
      code_folding: hide
      keep_md: true
bibliography: style/Pathogen_diagnostics.bib
csl: style/springer-basic-improved-author-date-with-italic-et-al-period.csl
---







# Objective
Develop highly specific, real-time chickpea fungal pathogens detection and analysis pipeline based on long-read Nanopore DNA sequencing 

# Aims
1.	Establish and optimise the methods and protocols for reproducible extraction of high molecular weight DNA suitable for long-read whole-genome-sequencing from fungal cultures and inoculated plant samples
2.	Develop target species' databases that will be used as references during sequencing (_Ascochyta rabiei_, _Botrytis cinerea_, _Botrytis fabae_ and potentially other closely related species)
3.	Accurately identify pathogens from mixed samples from controlled experiments and actual infected chickpea from growing paddocks (down to isolate-level resolution) 

# Experimental Design 
## Methods
* Extraction of high molecular weight pure fungal DNA suitable for long-read whole-genome-sequencing  
* Oxford Nanopore long-read sequencing, QC and taxonomic classification  
* Pathogen identification and abundance estimation  


Detailed methods, including code for running each of the analyses steps are provided in this repository.

